SwiftUIDZ/SwiftUIDZ/Helpers/LockerSlider.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  LockerSlider.swift
+//  SwiftUIDZ
+//
+//  Created by Алена Панченко on 01.02.2023.
+//
+
+import SwiftUI
+
+struct LockerSlider<V>: View where V : BinaryFloatingPoint, V.Stride : BinaryFloatingPoint {
+
+    // MARK: - Value
+    // MARK: Private
+    @Binding private var value: V
+    private let bounds: ClosedRange<V>
+    private let step: V.Stride
+
+    private let length: CGFloat    = 50
+    private let lineWidth: CGFloat = 2
+
+    @State private var ratio: CGFloat   = 0
+    @State private var startX: CGFloat? = nil
+
+
+    // MARK: - Initializer
+    init(value: Binding<V>, in bounds: ClosedRange<V>, step: V.Stride = 1) {
+        _value  = value
+    
+        self.bounds = bounds
+        self.step   = step
+    }
+
+
+    // MARK: - View
+    // MARK: Public
+    var body: some View {
+        GeometryReader { proxy in
```

Comment:
```suggestion
        GeometryReader { proxy in
```
разбей на маленькие вью